### ISSUE TITLE
Parameter(`duration_period`) returns unsorted years

### ISSUE THREAD

**Issue by OFR-IIASA** [iiasa/message_ix#485]
The parameter `duration_period` is not returned as a sorted list (by the years). 

NOTE: This could also apply to other parameters/sets etc. 

#### Code sample or context

```python

# after loading a scenario as `scen`
scen.par(`duration_period`)

# This yielded
    year  value unit
0   1970   10.0    y
1   1980   10.0    y
2   1990   10.0    y
3   1995    5.0    y
4   2000    5.0    y
5   2005    5.0    y
6   2010    5.0    y
7   2015    5.0    y
8   2020    5.0    y
9   2025    5.0    y
10  2030    5.0    y
11  2035    5.0    y
12  2040    5.0    y
13  2045    5.0    y
14  2050    5.0    y
15  2060   10.0    y
16  2070   10.0    y
17  2080   10.0    y
18  2090   10.0    y
19  2100   10.0    y
20  2110   10.0    y
21  1960   10.0    y
```

#### Expected

The dataframe should be sorted by the years

#### Problem description
Some other functions, such as `years_active` can return the wrong years as these do not specifically sort the retrieved dataframe.

**Comment by khaeru** [comment]
Some thoughts here:

1. The Java code in ixmp_source, and the ixmp JDBCBackend and general Backend API, have not ever advertised that the data returned will be sorted. So that is not something that should be expected by code (in message_ix, or user code).
2. IMHO the bug here is that `message_ix.Scenario.years_active()` does not sort the returned values: https://github.com/iiasa/message_ix/blob/56611adfda8fb8e88b5f4bdb4ab813e0ef894318/message_ix/core.py#L496-L497
   This is easily fixed:
   ```python
   data = self.par("duration_period").sort_values("year")
   ```
   Plus, perhaps, expand the docs to explicitly state point (1) above.
3. If the expectation is really "the dataframe [returned by `Scenario.par()`] should be sorted by the years", that's much more extensive and requires further discussion:
   - It's not a bug, but a change in behaviour; and not in message_ix, but in ixmp.
   - What should happen to parameters with 2 or more dimensions indexed by `year`, or without a `year` dimension?
   - Should there even be *any* fixed precedence of dimensions? User code might legitimately want data sorted by different dimensions; user code should ensure sortedness when it needs it, and be tolerant of unsortedness otherwise (e.g. using pandas group-by).
   - What is the performance impact? I am not sure how ixmp_source (Java) handles updates to large blocks of data. Suppose a parameter like `input` with 10⁶ rows; the user adds 1 row (`Scenario.add_par(…)`) then retrieves the new data. Storage and retrieval are fast if that single row is simply appended; slower if all data for the parameter has to be re-sorted. If the user chooses to write code that calls `add_par()` many times, then any automatic sorting will happen many times.

**Comment by OFR-IIASA** [comment]
@khaeru 
1. I was actually wondering whether this "was" once a feature (despite not being advertised).
2. I agree the fix is easy to implement, and can be the first simple solution to the observed issue.
3. I think it would be good to investigate/discuss this. Maybe it isn't a priority if indeed this was never a "feature", and hence this is only a single case where a feature assumes that years are sorted.

**Comment by LauWien** [comment]
This issue is resolved by the merge of #491 as @OFR-IIASA stated in the PR.
